File: week1/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    int count = 0;
    float cash = 0;
    do
        // Checks its a float > 0
    {
        // initially get the amount of change
        cash = get_float("Please input change owed:\n");
    }
    while (cash < 0.0);
    int cents = round(cash * 100);
    while (cents > 0)
    {
        //A series of while loops subtract each
        // coin tyoe starting with the largets
        while (cents >= 25)
        {
            cents -= 25;
            count ++;
        }
        while (cents >= 10)
        {
            cents -= 10;
            count ++;
        }
        while (cents >= 5)
        {
            cents -= 5;
            count ++;
        }
        while (cents >= 1)
        {
            cents -= 1;
            count ++;
        }
        printf("%i\n", count);
    }
}

File: week6/cash.py
from cs50 import get_float

def main():
    while True:
    # this do/wile loop from https://cs50.stackexchange.com/questions/32589/cs50-pset6-cash-py-not-returning-number-of-coins-used-for-change
        cash = get_float("Change owed: ")
        if cash >= 0:
            break
    cents = round(cash * 100)
    print(change(cents))

def change(cents):
    # based on my implementation of chash.c
    count = 0
    while cents > 0:
        if cents >= 25:
            cents -= 25
            count += 1
        elif cents >= 10:
            cents -= 10
            count += 1
        elif cents >= 5:
            cents -= 5
            count += 1
        elif cents >= 1:
            cents -= 1
            count += 1
    return count

main()